### DISCUSSION TITLE
want to know the use of this??

### DISCUSSION THREAD

**Discussion by 23f2005639** [mrdbourke/pytorch-deep-learning#360] [Q&A] [answered]
as I have used the forward function , what is the use of this(marked in image ) in forward function
![Screenshot 2023-03-19 190430](https://user-images.githubusercontent.com/58948510/226178629-5da89363-228b-424e-9c6c-dd08a244d26a.png)

**Comment by mrdbourke** [comment, marked as answer]
Hi @maxsn2005,

That is called a "type annotation" or "type hints" in Python.

```python
def forward(self, x: torch.Tensor) -> torch.Tensor:
    return self.weights * x + self.bias
```

The `x: torch.Tensor` means that the `forward()` method is expecting `x` to be of type `torch.Tensor`. 

And the `-> torch.Tensor` means that the `forward()` method returns a `torch.Tensor` too.

Type hints/annotations are helpful for when you'd like to know the inputs/outputs of your code/functions.

They don't necessarily return errors but when looking up documentation for what code does, the type annotation/hint shows you what the code is expecting as input/output.

You can see more here: https://realpython.com/python-type-checking/ 

Or here: https://docs.python.org/3/library/typing.html
